### DISCUSSION TITLE
How to compile multiple modules together?

### DISCUSSION THREAD

**Discussion by anandology** [austral/austral#416] [Q&A]
I'm trying to understand how modules work in Austral. I couldn't get an example to work.

I have a square module defined in two files `square.aui` and `square.aum`. I also have a `main.aum` with a main function. 

Source of the files is available in this gist.
https://gist.github.com/anandology/c8ff9836e541da642c1645a3d20abb1f

When I try to compile the `main.aum` alone, it complained that square module is not found.

```
$ austral compile main.aum --entrypoint=Main:main --output=main
Compiler call tree printed to calltree.html
Error:
  Title: Generic Error
  Module:
    Main
  Location:
    [no span available]
  Description:
    No such module.
  Code:
    [no span available]
```

And I couldn't find a way to pass multiple files.  The `austral` command shows it is possible to pass both interface and body files.

```
$ austral compile --help
austral compile
...
Positional arguments:
    module    Of the form 'file.aui:file.aum' for modules with
              both an interface and body file, or 'file.aum' for
              modules with only a body.
```

But passing both failed with error.

```
$ austral compile square.aui:square.aum --entrypoint=Main:main --output=main
Fatal error: exception Sys_error("square.aui:square.aum: No such file or directory")
```

Is this a feature that is yet to be implemented? 

How do I compile all these three files together?

**Comment by eudoxia0** [comment]
So the bootstrapping compiler is a very simple whole-program compiler. You need to pass every module manually, in order:

```
$ austral compile \
    src/A.aui,src/A.aum \
    src/B.aui,src/B.aum \
    src/C.aui,src/C.aum \
    --entrypoint=C:main \
    --output=program
```

I think the error message here is wrong, it's showing a colon instead of a comma.
